src/redux/slices/userSlice.js

```diff
@@ -70,6 +71,52 @@ export const getUserDetails = createAsyncThunk(
   }
 );
 
+export const getAllBookmarkedPosts = createAsyncThunk(
+  "posts/getAllBookmarkedPosts",
+  async (_, { rejectWithValue }) => {
+    try {
+      const response = await axios({
+        method: "get",
+        url: `/api/users/bookmark/`,
```

Comment:
suggestion: you can create a folder with specific files for each slice, & move all these axios requests.

Reply:
will do it later